frontend/src/employee-frontend/state/child.tsx

```diff
@@ -2,220 +2,141 @@
 }) {
-  const [person, setPerson] = useState<Result<PersonJSON>>(defaultState.person)
-  const [permittedActions, setPermittedActions] = useState(
-    defaultState.permittedActions
-  )
-  const [serviceNeeds, setServiceNeeds] = useState<Result<ServiceNeed[]>>(
-    defaultState.serviceNeeds
+  const { roles } = useContext(UserContext)
+
+  const [childResponse, setChildResponse] = useState<Result<ChildResponse>>(
+    Loading.of()
   )
-  const [serviceNeedOptions, setServiceNeedOptions] = useState<
-    Result<ServiceNeedOption[]>
-  >(defaultState.serviceNeedOptions)
-  const [assistanceNeeds, setAssistanceNeeds] = useState<
-    Result<AssistanceNeed[]>
-  >(defaultState.assistanceNeeds)
-  const [assistanceActions, setAssistanceActions] = useState<
-    Result<AssistanceAction[]>
-  >(defaultState.assistanceActions)
-  const [additionalInformation, setAdditionalInformation] = useState<
-    Result<AdditionalInformation>
-  >(defaultState.additionalInformation)
-  const [feeAlterations, setFeeAlterations] = useState<Result<FeeAlteration[]>>(
-    defaultState.feeAlterations
+  const [permittedActions, setPermittedActions] = useState<
+    Set<Action.Child | Action.Person>
+  >(emptyPermittedActions)
+
+  const setFullChildResponse = useCallback(
+    (response: Result<ChildResponse>) => {
+      setPermittedActions(
+        response
+          .map(
+            ({ permittedActions, permittedPersonActions }) =>
+              new Set([...permittedActions, ...permittedPersonActions])
+          )
+          .getOrElse(emptyPermittedActions)
+      )
+      setChildResponse(response)
+    },
+    []
   )
-  const [placements, setPlacements] = useState<
-    Result<DaycarePlacementWithDetails[]>
-  >(defaultState.placements)
-  const [parentships, setParentships] = useState<Result<Parentship[]>>(
-    defaultState.parentships
+  const loadChild = useRestApi(getChildDetails, setFullChildResponse)
+  useEffect(() => loadChild(id), [loadChild, id])
+
+  const person = useMemo(
+    () => childResponse.map((response) => response.person),
+    [childResponse]
   )
-  const [backupCares, setBackupCares] = useState<Result<ChildBackupCare[]>>(
-    defaultState.backupCares
+  const setPerson = useCallback(
+    (person: PersonJSON) =>
+      setChildResponse((prev) =>
+        prev.map((child) => ({
+          ...child,
+          person
+        }))
+      ),
+    []
   )
-  const [guardians, setGuardians] = useState<Result<PersonJSON[]>>(
-    defaultState.guardians
+
+  const [placements, loadPlacements] = useApiState(
+    () => getPlacements(id),
+    [id]
   )
-  const [applications, setApplications] = useState<
-    Result<PersonApplicationSummary[]>
-  >(defaultState.applications)
-  const [vasus, setVasus] = useState<Result<VasuDocumentSummary[]>>(
-    defaultState.vasus
+  const [parentships] = useApiState(
+    async (): Promise<Result<Parentship[]>> =>
+      requireRole(roles, 'SERVICE_WORKER', 'UNIT_SUPERVISOR', 'FINANCE_ADMIN')
```

Comment:
Change this to a `permittedActions` check now that you touched it 😁

Reply:
Not sure how to do that, it needs some investigation. The problem is that while it's easy to check whether someone is a `UNIT_SUPERVISOR` for a child, it's not really supported for adults. So if I change the check in the service to be based on actions, unit supervisors stop seeing parentships for adults.

Reply:
Ah, I didn't realize that service was still doing role checks. It could do different checks depending on the parameter but that's getting really far away from the scope of this PR so no need to do it.